File: Semester 5/Mobile Applications/lab02/Exercise.kt
package com.example.lab01.data

import androidx.compose.runtime.Immutable

@Immutable
data class Exercise(
    val id: Int,
    val name: String,
    val description: String,
    val category: String,
    val primaryMuscles: List<String>,
    val secondaryMuscles: List<String>,
    val equipment: List<String>,
    val image: Int
)

File: Semester 5/Mobile Applications/lab05/server/app/models/exercise.rb
class Exercise < ApplicationRecord
    validates :name, presence: { message: 'cannot be blank or contain only whitespace' }
    validates :description, presence: { message: 'cannot be blank or contain only whitespace' }
    validates :category, presence: { message: 'cannot be blank or contain only whitespace' }
    validates :primary_muscle, presence: { message: 'cannot be blank or contain only whitespace' }
    validates :secondary_muscles, presence: { message: 'cannot be blank or contain only whitespace' }
    validates :equipment, presence: { message: 'cannot be blank or contain only whitespace' }
end
